### PR TITLE
[exporter/clickhouse] use generated status header

### PR DESCRIPTION
**Pull request by codeboten**
This PR updates the component to generate the status table using mdatagen.

Linked issue: #21213

### CODE REVIEW

File: exporter/clickhouseexporter/factory.go [hunk 1]
```diff
@@ -10,23 +12,20 @@ import (
...
 
-const (
-	// The value of "type" key in configuration.
-	typeStr = "clickhouse"
-	// The stability level of the exporter.
-	stability = component.StabilityLevelAlpha
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/clickhouseexporter/internal/metadata"
 )
 
+const ()
```
Comment (crobert-1) [inline exporter/clickhouseexporter/factory.go:19]: Is `const ()` still required?

Comment (codeboten) [reply exporter/clickhouseexporter/factory.go:19]: It is not. Accidentally left it behind!

Comment (codeboten) [inline exporter/clickhouseexporter/factory.go:19]:
```suggestion
```

File: exporter/clickhouseexporter/factory.go [hunk 2]
```diff
@@ -10,23 +12,19 @@ import (
...
 
-const (
-	// The value of "type" key in configuration.
-	typeStr = "clickhouse"
-	// The stability level of the exporter.
-	stability = component.StabilityLevelAlpha
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/clickhouseexporter/internal/metadata"
 )
 
+
```
Comment (codeboten) [inline exporter/clickhouseexporter/factory.go:19]:
```suggestion
```

**Review by crobert-1 — APPROVED** [review]

**Review by atoulme — APPROVED** [review]
Thanks Curtis for the review!

**Review by dmitryax — APPROVED** [review]

**Pull request merged**